packages/expo-contacts/android/src/main/java/expo/modules/contacts/ContactsModule.java

```diff
@@ -76,18 +76,46 @@ public void onCreate(ModuleRegistry moduleRegistry) {
     }
   };
 
+  @ExpoMethod
+  public void requestPermissionsAsync(final Promise promise) {
+    Permissions permissionsManager = mModuleRegistry.getModule(Permissions.class);
+    if (permissionsManager == null) {
+      promise.reject("E_NO_PERMISSIONS", "Permissions module is null. Are you sure all the installed Expo modules are properly linked?");
+      return;
+    }
+    if (permissionsManager.isPermissionPresentInManifest(Manifest.permission.WRITE_CALENDAR)) {
+      permissionsManager.askForPermissionsWithPromise(promise, Manifest.permission.READ_CONTACTS, Manifest.permission.WRITE_CONTACTS);
+    } else {
+      permissionsManager.askForPermissionsWithPromise(promise, Manifest.permission.READ_CONTACTS);
+    }
+  }
+
+  @ExpoMethod
+  public void getPermissionsAsync(final Promise promise) {
+    Permissions permissionsManager = mModuleRegistry.getModule(Permissions.class);
+    if (permissionsManager == null) {
+      promise.reject("E_NO_PERMISSIONS", "Permissions module is null. Are you sure all the installed Expo modules are properly linked?");
+      return;
+    }
+    if (permissionsManager.isPermissionPresentInManifest(Manifest.permission.WRITE_CALENDAR)) {
+      permissionsManager.getPermissionsWithPromise(promise, Manifest.permission.READ_CONTACTS, Manifest.permission.WRITE_CONTACTS);
+    } else {
+      permissionsManager.getPermissionsWithPromise(promise, Manifest.permission.READ_CONTACTS);
+    }
+  }
+
   // TODO: Evan: Test
   @ExpoMethod
   public void getContactsAsync(final Map<String, Object> options, final Promise promise) {
-    if (isMissingPermissions(promise)) return;
+    if (isMissingReadPermission(promise)) return;
```

Comment:
```suggestion
    if (isMissingReadPermission(promise)) {
      return;
    }
```
In general we are against conditional one liners, same goes for lack of `{``}`